store/src/java/com/zimbra/cs/html/owasp/OwaspDefang.java

```diff
@@ -55,12 +55,28 @@ public void defang(InputStream is, boolean neuterImages, Writer out) throws IOEx
     @Override
     public void defang(Reader reader, boolean neuterImages, Writer out) throws IOException {
         String html = CharStreams.toString(reader);
-        String sanitizedHtml = runSanitizer(html, neuterImages);
+        // temporary fix to check extra comment tag & remove
+        String formatHtml = checkExtraCommentLine(html);
+        String sanitizedHtml = runSanitizer(formatHtml, neuterImages);
         if (sanitizedHtml != null) {
             out.write(sanitizedHtml);
         }
         out.close();
     }
+    
+    // method to check extra comment tag .. if comment "<!--" has started but "-->" not ended we are skipping unbalanced comment tag
+    private String checkExtraCommentLine(String html) {
+    	String[] arrOfStr = html.split("<!--");
+    	String contain = "";
+    	for (String a : arrOfStr) {
+    		if( a.contains("-->")) {
```

Comment:
this logic of split html and just check for one occurrence of -->, would be problem. For e.g.
<!--
some lines
<!--
some lines
-->